### ISSUE TITLE
Content entries push error with localised select

### ISSUE THREAD

**Issue by clecoufl** [locomotivecms/wagon#343]
Hi @did,

There seems to be a bug with wagon when pushing content entries that contain a localized select field.

**Wagon version** : 2.3.0
**Engine version** : 3.3.0

Content type select definition (partial): 
```
...
- section_category:
    label: 'Catégorie'
    type: select
    required: false
    hint: 'Introduction, Strategie, Design, Experience, Témoignage...'
    localized: true
    select_options:
      en: ['Introduction', 'Strategy', 'Design', 'Experience', 'Testimonial']
      fr: ['Introduction', 'Strategie', 'Design', 'Experience', 'Témoignage']
...
```

Corresponding data:
```
- Images typo:
    _slug:
      en: brand-design
      fr: brand-design
    case_study: brandcase
    internal_name: Images typo
    over_title:
      en: brand design
      fr: brand design
    title:
      en: brand design
      fr: brand design
    section_category:
      en: Strategy
      fr: Strategie
    display_mode: sans_titres
- Brand Design:
    _slug:
      en: brand-design-2
      fr: brand-design-2
    case_study: brandcase2
    internal_name: Brand Design
    over_title:
      en: Brand Design
      fr: Brand Design
    title:
      en: Brand Design
      fr: Brand Design
    section_category:
      en: Design
      fr: Design
    display_mode: sans_titres
```

The first one is perfectly persisted to the Locomotive engine.
The second one fails with the following stack trace : 

```
# Error description:
undefined method `section_category_id=' for #<Locomotive::API::Forms::ContentEntryForm:0x00007f18ea84c0b0>

# Backtrace:

        /usr/local/bundle/gems/locomotivecms_coal-1.4.0/lib/locomotive/coal/resources/concerns/request.rb:36:in `do_request'
        /usr/local/bundle/gems/locomotivecms_coal-1.4.0/lib/locomotive/coal/resources/concerns/request.rb:15:in `put'
        /usr/local/bundle/gems/locomotivecms_coal-1.4.0/lib/locomotive/coal/resources/base.rb:35:in `update_with_locale'
        /usr/local/bundle/gems/locomotivecms_wagon-2.3.12/lib/locomotive/wagon/commands/push_sub_commands/push_content_entries_command.rb:48:in `persist'
        /usr/local/bundle/gems/locomotivecms_wagon-2.3.12/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:34:in `block in _push'
        /usr/local/bundle/gems/locomotivecms_wagon-2.3.12/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:30:in `each'
        /usr/local/bundle/gems/locomotivecms_wagon-2.3.12/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:30:in `_push'
        /usr/local/bundle/gems/locomotivecms_wagon-2.3.12/lib/locomotive/wagon/commands/push_sub_commands/push_content_entries_command.rb:12:in `block in _push'
        /usr/local/bundle/gems/locomotivecms_wagon-2.3.12/lib/locomotive/wagon/commands/push_sub_commands/push_content_entries_command.rb:10:in `each'
        /usr/local/bundle/gems/locomotivecms_wagon-2.3.12/lib/locomotive/wagon/commands/push_sub_commands/push_content_entries_command.rb:10:in `_push'
        /usr/local/bundle/gems/locomotivecms_wagon-2.3.12/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:25:in `block in _push_with_timezone'
        /usr/local/bundle/gems/activesupport-4.2.10/lib/active_support/core_ext/time/zones.rb:48:in `use_zone'
        /usr/local/bundle/gems/locomotivecms_wagon-2.3.12/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:24:in `_push_with_timezone'
        /usr/local/bundle/gems/locomotivecms_wagon-2.3.12/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:18:in `block in push'
        /usr/local/bundle/gems/activesupport-4.2.10/lib/active_support/notifications.rb:166:in `instrument'
        /usr/local/bundle/gems/locomotivecms_wagon-2.3.12/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:53:in `instrument'
        /usr/local/bundle/gems/locomotivecms_wagon-2.3.12/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:16:in `push'
        /usr/local/bundle/gems/locomotivecms_wagon-2.3.12/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:12:in `push'
        /usr/local/bundle/gems/locomotivecms_wagon-2.3.12/lib/locomotive/wagon/commands/push_command.rb:55:in `block in _push'
        /usr/local/bundle/gems/locomotivecms_wagon-2.3.12/lib/locomotive/wagon/commands/push_command.rb:84:in `block in each_resource'
        /usr/local/bundle/gems/locomotivecms_wagon-2.3.12/lib/locomotive/wagon/commands/push_command.rb:77:in `each'
        /usr/local/bundle/gems/locomotivecms_wagon-2.3.12/lib/locomotive/wagon/commands/push_command.rb:77:in `each_resource'
        /usr/local/bundle/gems/locomotivecms_wagon-2.3.12/lib/locomotive/wagon/commands/push_command.rb:54:in `_push'
        /usr/local/bundle/gems/locomotivecms_wagon-2.3.12/lib/locomotive/wagon/commands/push_command.rb:41:in `push'
        /usr/local/bundle/gems/locomotivecms_wagon-2.3.12/lib/locomotive/wagon/commands/push_command.rb:31:in `push'
        /usr/local/bundle/gems/locomotivecms_wagon-2.3.12/lib/locomotive/wagon.rb:73:in `push'
        /usr/local/bundle/gems/locomotivecms_wagon-2.3.12/lib/locomotive/wagon/cli.rb:318:in `deploy'
```

When looking at the entity in the **_push** method of wagon/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb
it appears that the select field is not parsed the same way for both content entries : 
First one :
```
{"_id":"brand-design","_slug":{"en":"brand-design","fr":"brand-design"},"_visible":true,"_position":3,"content_type_slug":"case_study_sections","created_at":"2018-03-02T18:20:17.584+01:00","updated_at":"2018-03-02T18:20:17.584+01:00","_label":"Images typo","case_study_id":"brandcase","internal_name":"Images typo","over_title":{"en":"brand design","fr":"brand design"},"title":{"en":"brand design","fr":"brand design"},"section_category_id":{"en":"Strategy","fr":null},"display_mode_id":"sans_titres","header_image":{"url":null,"filename":null,"size":null,"updated_at":"2018-03-02T18:20:17.584+01:00"}}

Select corresponding content: 
 "section_category_id":{"en":"Strategy","fr":null}
```

Second one - failing -: 
```
{"_id":"brand-design-2","_slug":{"en":"brand-design-2","fr":"brand-design-2"},"_visible":true,"_position":4,"content_type_slug":"case_study_sections","created_at":"2018-03-02T18:20:17.584+01:00","updated_at":"2018-03-02T18:20:17.584+01:00","_label":"Brand Design","case_study_id":"brandcase2","internal_name":"Brand Design","over_title":{"en":"Brand Design","fr":"Brand Design"},"title":{"en":"Brand Design","fr":"Brand Design"},"section_category_id":{"en":"Design","fr":"Design"},"display_mode_id":"sans_titres","header_image":{"url":null,"filename":null,"size":null,"updated_at":"2018-03-02T18:20:17.584+01:00"}}

Select corresponding content:
"section_category_id":{"en":"Design","fr":"Design"}
```

en being the default language, whenever the name of the select in the secondary language (fr) is identical to its name in the default language, the entity is filled with a value for select_name_id for the secondary language instead of null.

I managed to have the push working again by having a different value for fr than for en for all localized select fields of the data : ex : 
```
section_category:
  en: Design
  fr: Design-fr
...
section_category:
  en: Experience
  fr: Experience-fr
etc...
```

In short, the bug happens whenever trying to push a content entry that has a localized select with the exact same value in secondary language(s) as in the default language.

**Comment by DonKoko** [comment]
Just want to mention I was having the same issue. @manuchap found this solution to be able to push:

> Solved it by adding _option at the end of the localized field:
> ```
> test_option:
>       fr: Valeur 1
>       en: Value 1
> ```

So in your case it should be `section_category_option`. 
Hope it helps.

**Comment by clecoufl** [comment]
> Just want to mention I was having the same issue. @manuchap found this solution to be able to push:
> 
> > Solved it by adding _option at the end of the localized field:
> > ```
> > test_option:
> >       fr: Valeur 1
> >       en: Value 1
> > ```
> 
> So in your case it should be `section_category_option`.
> Hope it helps.

Thank you @DonKoko for the tip.
Unfortunately it did not work out for me.
**WAGON VERSION** : 2.4.1
**ENGINE VERSION** : 3.4.1

I built a sample content entry with one simple select field (one translation of which being identical in both languages) : 
app/content_types/translated_selects.yml
```
[...]
  - myfield_option:
      label: 'Select option'
      type: select
      required: false
      localized: true
      select_options:
        en: ['Option A', 'Option B', 'Option C']
        fr: ['Option A_fr', 'Option B_fr', 'Option C']
[...]
```
And corresponding test data.
data/translated_selects.yml
```
- Entry A:
    _slug:
      en: entry-a
      fr: entry-a
    name: Entry A
    myfield_option:
      en: Option A
      fr: Option A_fr
- Entry B:
    _slug:
      en: entry-b
      fr: entry-b
    name: Entry B
    myfield_option:
      en: Option B
      fr: Option B_fr
- Entry C:
    _slug:
      en: entry-c
      fr: entry-c
    name: Entry C
    myfield_option:
      en: Option C
      fr: Option C
```

When I push wagon with data (bundle exec wagon deploy --data -v), I still get the error : 
```
persisting Translated selects / Entry A [done]
  persisting Translated selects / Entry B [done]
  persisting Translated selects / Entry C [done]
  persisting Translated selects / Entry A in fr [skip]
  persisting Translated selects / Entry B in fr [skip]
  persisting Translated selects / Entry C in fr [failed]
    Locomotive Back-office error. Contact your administrator or check your application logs.

# Error description:
unknown attribute 'myfield_option_id' for Locomotive::API::Forms::ContentEntryForm.

# Backtrace:

        /usr/local/bundle/gems/locomotivecms_coal-1.5.1/lib/locomotive/coal/resources/concerns/request.rb:36:in `do_request'
        /usr/local/bundle/gems/locomotivecms_coal-1.5.1/lib/locomotive/coal/resources/concerns/request.rb:15:in `put'
        /usr/local/bundle/gems/locomotivecms_coal-1.5.1/lib/locomotive/coal/resources/base.rb:35:in `update_with_locale'
        /usr/local/bundle/gems/locomotivecms_wagon-2.4.1/lib/locomotive/wagon/commands/push_sub_commands/push_content_entries_command.rb:42:in `persist'
        /usr/local/bundle/gems/locomotivecms_wagon-2.4.1/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:34:in `block in _push'
        /usr/local/bundle/gems/locomotivecms_wagon-2.4.1/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:30:in `each'
        /usr/local/bundle/gems/locomotivecms_wagon-2.4.1/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:30:in `_push'
        /usr/local/bundle/gems/locomotivecms_wagon-2.4.1/lib/locomotive/wagon/commands/push_sub_commands/push_content_entries_command.rb:12:in `block in _push'
        /usr/local/bundle/gems/locomotivecms_wagon-2.4.1/lib/locomotive/wagon/commands/push_sub_commands/push_content_entries_command.rb:10:in `each'
        /usr/local/bundle/gems/locomotivecms_wagon-2.4.1/lib/locomotive/wagon/commands/push_sub_commands/push_content_entries_command.rb:10:in `_push'
        /usr/local/bundle/gems/locomotivecms_wagon-2.4.1/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:25:in `block in _push_with_timezone'
        /usr/local/bundle/gems/activesupport-5.1.6.2/lib/active_support/core_ext/time/zones.rb:64:in `use_zone'
        /usr/local/bundle/gems/locomotivecms_wagon-2.4.1/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:24:in `_push_with_timezone'
        /usr/local/bundle/gems/locomotivecms_wagon-2.4.1/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:18:in `block in push'
        /usr/local/bundle/gems/activesupport-5.1.6.2/lib/active_support/notifications.rb:168:in `instrument'
        /usr/local/bundle/gems/locomotivecms_wagon-2.4.1/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:50:in `instrument'
        /usr/local/bundle/gems/locomotivecms_wagon-2.4.1/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:16:in `push'
        /usr/local/bundle/gems/locomotivecms_wagon-2.4.1/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:12:in `push'
        /usr/local/bundle/gems/locomotivecms_wagon-2.4.1/lib/locomotive/wagon/commands/push_command.rb:55:in `block in _push'
        /usr/local/bundle/gems/locomotivecms_wagon-2.4.1/lib/locomotive/wagon/commands/push_command.rb:84:in `block in each_resource'
        /usr/local/bundle/gems/locomotivecms_wagon-2.4.1/lib/locomotive/wagon/commands/push_command.rb:77:in `each'
        /usr/local/bundle/gems/locomotivecms_wagon-2.4.1/lib/locomotive/wagon/commands/push_command.rb:77:in `each_resource'
        /usr/local/bundle/gems/locomotivecms_wagon-2.4.1/lib/locomotive/wagon/commands/push_command.rb:54:in `_push'
        /usr/local/bundle/gems/locomotivecms_wagon-2.4.1/lib/locomotive/wagon/commands/push_command.rb:41:in `push'
        /usr/local/bundle/gems/locomotivecms_wagon-2.4.1/lib/locomotive/wagon/commands/push_command.rb:31:in `push'
        /usr/local/bundle/gems/locomotivecms_wagon-2.4.1/lib/locomotive/wagon.rb:73:in `push'
        /usr/local/bundle/gems/locomotivecms_wagon-2.4.1/lib/locomotive/wagon/cli.rb:318:in `deploy'
```
@did : I attached  zips of sample content type files (definition and data) if needed. 
[translated_selects_with_option_suffix.zip](https://github.com/locomotivecms/wagon/files/3005016/translated_selects_with_option_suffix.zip)
[translated_selects.zip](https://github.com/locomotivecms/wagon/files/3005017/translated_selects.zip)

**BTW** : When I change the name of the select field (i.e from myfield_option to myfield) and try to push again, I get an Internal server error from the engine caused by a SystemStackError (stack level too deep) 
```
SystemStackError (stack level too deep):                                                                                                                                          FATAL -- : [c040ac44-b621-4bca-9621-b18d12e6d72a] activesupport (5.1.6.2) lib/active_support/core_ext/object/json.rb:163:in `dup'                                                                                                          
        activesupport (5.1.6.2) lib/active_support/core_ext/object/json.rb:163:in `block in as_json'
        activesupport (5.1.6.2) lib/active_support/core_ext/object/json.rb:163:in `each'
        activesupport (5.1.6.2) lib/active_support/core_ext/object/json.rb:163:in `map'
        activesupport (5.1.6.2) lib/active_support/core_ext/object/json.rb:163:in `as_json'
        activesupport (5.1.6.2) lib/active_support/core_ext/object/json.rb:144:in `block in as_json'
        activesupport (5.1.6.2) lib/active_support/core_ext/object/json.rb:144:in `map'
        [...etc....]
```
I had to delete the content entry on the engine before being able to push again (bundle exec wagon delete MYENV content_entry translated_selects)
Would you like me to create a new issue for this problem?

**Comment by did** [comment]
closing it because I fixed a couple of issue related to localized select fields
